web/web/models/meter.py

```diff
@@ -96,10 +98,31 @@ def __repr__(self):
         return f'<Meter meter_id={self.meter_id} is_active={self.is_active}>'
 
 
-#marshmallow schema
 class MeterSchema(SQLAlchemyAutoSchema):
+    meter_type = fields.Method("get_meter_type", deserialize="load_meter_type")
+    map_location = fields.Method("get_map_location", deserialize="load_map_location")
+    postal_code = fields.Method("get_postal_code")
+
+    def get_postal_code(self, obj):
+        return obj.service_location.address.postal_code
+
+    def get_map_location(self, obj):
+        return obj.service_location.map_location
+
+    def load_map_location(self, value):
+        return
+
+    def get_meter_type(self, obj):
+        return obj.meter_type.value
+
+    def load_meter_type(self, value):
+        meter_enum = MeterType.check_value(value)
+        if not meter_enum:
+            raise ValidationError(f'{value} is an invalid meter type')
+        return meter_enum
+
     class Meta:
         model = Meter
         include_relationships = True
         load_instance = True
-        include_fk = True
\ No newline at end of file
+        include_fk = True
```

Comment:
@avpeery checkout the updated schema here. We should try to overload the schema with the responsibility to create the ingest format and output responses as much as possible. Also, this should be where we consolidate all the validation per model, since we should try to use the schema as much as possible to marshal data back and forth.

Reply:
Hey thanks so much for your work on this - I totally see where you are going, and it makes a lot more sense to me now. This looks good to me